<!-- -*- mode: XML -*- -->
<launch>
  <arg name="input_topic" default="/LidarAll" />
  <arg name="baselink_support" default="false" />
  <arg name="reproduce_result_mode" default="false" />

  <arg name="score_threshold_car" default="0.4" />
  <arg name="nms_overlap_threshold_car" default="0.05" />

  <arg name="score_threshold_ped_cyc" default="0.2" />
  <arg name="nms_overlap_threshold_ped_cyc" default="0.05" />

<!-- car -->
  <node pkg="lidar_point_pillars" type="lidar_point_pillars_car" name="lidar_point_pillars_car" output="screen" respawn="true">
    <remap from="/points_raw" to="$(arg input_topic)"/>
    <param name="baselink_support"            value="$(arg baselink_support)" />
    <param name="reproduce_result_mode"        value="$(arg reproduce_result_mode)" />
    <param name="score_threshold"   value="$(arg score_threshold_car)" />
    <param name="nms_overlap_threshold"         value="$(arg nms_overlap_threshold_car)" />
    <param name="pfe_onnx_file"   value="$(find lidar_point_pillars)/model/car/pfe.onnx" />
    <param name="rpn_onnx_file"   value="$(find lidar_point_pillars)/model/car/rpn.onnx" />
  </node>

<!-- ped & cyc -->
  <node pkg="lidar_point_pillars" type="lidar_point_pillars_ped_cyc" name="lidar_point_pillars_ped_cyc" output="screen" respawn="true">
    <remap from="/points_raw" to="$(arg input_topic)"/>
    <param name="baselink_support"            value="$(arg baselink_support)" />
    <param name="reproduce_result_mode"        value="$(arg reproduce_result_mode)" />
    <param name="score_threshold"   value="$(arg score_threshold_ped_cyc)" />
    <param name="nms_overlap_threshold"         value="$(arg nms_overlap_threshold_ped_cyc)" />
    <param name="pfe_onnx_file"   value="$(find lidar_point_pillars)/model/ped_cyc/pfe.onnx" />
    <param name="rpn_onnx_file"   value="$(find lidar_point_pillars)/model/ped_cyc/rpn.onnx" />
  </node>

<!-- integrator -->
  <node pkg="lidar_point_pillars" type="lidar_point_pillars_integrator" name="lidar_point_pillars_integrator" output="screen" respawn="true" />

</launch>
